plumpy/ports.py

```diff
@@ -370,12 +370,26 @@ def __init__(self,
                  validator=None,
                  valid_type=None,
                  default=UNSPECIFIED,
-                 dynamic=False):
+                 dynamic=False,
+                 populate_defaults=True):
+        """Construct a port namespace.
+
+        :param name: the name of the namespace
+        :param help: the help string
+        :param required: boolean, if True the validation will fail if no value is specified for this namespace
+        :param validator: an optional validator for the namespace
+        :param valid_type: optional tuple of valid types in the case of a dynamic namespace
+        :param default: default value for the port
+        :param dynamic: boolean, if True, the namespace will accept values even when no explicit port is defined
+        :param populate_defaults: boolean, if False, the pre-processing step does not populate defaults, also not of any
+            nested ports, if no value was explicitly specified for this namespace.
+        """
```

Comment:
I would just add an explicit note: ".. note: if the port is explicitly specified, this parameter is ignored and the defaults are populated"

Reply:
that is what it says, but maybe I can reword it, so that it more clear

Reply:
I have changed the docstring. I think it should be clear now